File: Service/Models/PublicWageringSnapshotModel.cs
namespace TecmoTourney.Models
{
    /// <summary>Read-only odds and market depth for a game; public, no authentication.</summary>
    public class PublicWageringSnapshotModel
    {
        public int GameResultId { get; set; }
        public int Player1Id { get; set; }
        public int Player2Id { get; set; }
        public string Player1Name { get; set; } = string.Empty;
        public string Player2Name { get; set; } = string.Empty;
        public int Player1ProfilePic { get; set; }
        public int Player2ProfilePic { get; set; }
        public PublicWageringOddsModel Odds { get; set; } = new();
        public BettableGameMarketDepthModel MarketDepth { get; set; } = new();
    }

    public class PublicWageringOddsModel
    {
        public decimal Spread { get; set; }
        public int? FavoredPlayerId { get; set; }
        public decimal? OverUnder { get; set; }
        public decimal? MoneyLinePlayer1 { get; set; }
        public decimal? MoneyLinePlayer2 { get; set; }
        public string Summary { get; set; } = string.Empty;
    }
}

File: Client/tecmo-tourney/src/app/core/models/public-wagering-snapshot.model.ts
export interface IPublicWageringOdds {
  spread: number;
  favoredPlayerId: number | null;
  overUnder: number | null;
  moneyLinePlayer1: number | null;
  moneyLinePlayer2: number | null;
  summary: string;
}

export interface IPublicWageringMarketDepth {
  maxMarketImbalance: number;
  spreadPlayer1: number;
  spreadPlayer2: number;
  over: number;
  under: number;
  moneyLinePlayer1: number;
  moneyLinePlayer2: number;
}

export interface IPublicWageringSnapshot {
  gameResultId: number;
  player1Id: number;
  player2Id: number;
  player1Name: string;
  player2Name: string;
  player1ProfilePic: number;
  player2ProfilePic: number;
  odds: IPublicWageringOdds;
  marketDepth: IPublicWageringMarketDepth;
}
